sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/AppConfigurationRefresh.java

```diff
@@ -25,7 +30,9 @@ public interface AppConfigurationRefresh extends ApplicationEventPublisherAware
      * Checks configurations to see if configurations should be reloaded. If the refresh interval has passed and a
      * trigger has been updated configuration are reloaded.
      */
-    void refreshAsync();
+    default void refreshAsync() {
```

Comment:
Why is this returning a `void` when it is an async call? It should return `Mono<Void>` if it has to be async.

Reply:
We have this exact same thing in another method that returns a mono. This is a work around to try and make this scenario work for Spring Flux customers. The implementation version of this just calls the same refresh code in another thread. If not, it can cause Spring Flux customers.